SPARCbot.py

```diff
@@ -168,8 +184,276 @@ async def iwantto(self, ctx, role: str):
     async def unsure(self, ctx):
         await timed_send(ctx, 'I\'m afraid I can\'t let you do that.')
 
+class Bets(commands.Cog):
 
 
+    def __init__(self,bot):
+        self.bot = bot
+        self.lock = False
+
+    def add_new_bet(self,name,statement,status='open'):
+        '''add a new bet'''
+        #load existing bets
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        current_bet_id = bet_log['current_bet_id'] + 1
+        bet_key = 'bet_'+str(current_bet_id)
+
+        #make the new bet
+        new_bet = {
+            "bet_id": current_bet_id,
+            "bidder": name,
+            "status": status,
+            "statement": statement
+        }
+        bet_log[bet_key] = new_bet
+
+        #update current_bet_id
+        bet_log['current_bet_id'] = current_bet_id
+
+        #save updated bets
+        with open("bet_log.json","w") as write_file:
+            json.dump(bet_log,write_file,indent=4)
+        return bet_log
+
+    def check_author(self,ctx,bet_id: int):
+        '''checks if the person giving commands is in the bet'''
+        #load bets
+        with open("bet_log.json", "r") as read_file:
+            bet_log = json.load(read_file)
+        bet_key = 'bet_'+str(bet_id)
+
+        #check bidder and seller (if it exists)
+        try:
+            if 'seller' in bet_log[bet_key]:
+                return bet_log[bet_key]['bidder'] == ctx.author.id or bet_log[bet_key]['seller'] == ctx.author.id
+            else:
+                return bet_log[bet_key]['bidder'] == ctx.author.id
+        except KeyError as e:
+            raise Exception('Couldn\'t check author')
+
+    @commands.command()
+    async def bet(self,ctx,*,statement):
+        '''<statement>: create an open bet'''
+        #share a unique bet_id
+        try:
+            #default bet is one-time
+            status = 'open'
+
+            #make an ongoing bet
+            if statement.startswith(config['bet_status']['standing']):
+                statement = statement.split(' ',1)[1]
+                status = 'standing'
+
+            #add the bet
+            bet_log = self.add_new_bet(ctx.author.id,statement,status)
+            bet_id = bet_log['current_bet_id']
+            nick = await get_nick_from_id(ctx,ctx.author.id)
+            await timed_send(ctx, nick+' added bet '+str(bet_id)+': \"'+statement+'\"')
+        except Exception as e:
+            await timed_send(ctx, 'Hmm...that didn\'t seem to work.')
+            raise
+
+    @commands.command()
+    async def imout(self,ctx):
+        '''Cancels your most recent unclaimed offer'''
+        if self.lock:
+            await timed_send(ctx, 'still working!')
+            return
+        self.lock = True
```

Comment:
why is it that this command uses the lock specifically? maybe all of them should use it, since there are file read/writes happening in all as far as I can tell?

Reply:
also there's probably a Pythonic (TM) way to do it lol which we should use as good practice. maybe https://docs.python.org/3.6/library/threading.html#threading.Lock

Reply:
gonna go with the dumb version for the sake of not spending too much time